overlord/snapstate/handlers.go

```diff
@@ -1061,6 +1061,49 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	// Notify link snap participants about link changes.
 	notifyLinkParticipants(t, snapsup.InstanceName())
+
+	// undo migration if appropriate
+	if snapsup.Flags.Revert {
```

Comment:
Nitpick, maybe reverse the condition to reduce the indentation?
```if !snapsup.Flags.Revert {
  return nil
}```

Reply:
The reason why I opted not to do that (and the reason for the `if err != nil` at the bottom instead of just returning) is to avoid later changes from having to change this block of code. I thought it was less error-prone

Reply:
Ok, sounds good, it's fine as is.